src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -331,6 +334,74 @@ struct ScalarOp<ONNXTanOp> {
   using IOp = NotSuportedScalarOp;
 };
 
+//===----------------------------------------------------------------------===//
+// Scalar unary ops for lowering ONNXGeluOp
+//===----------------------------------------------------------------------===//
+template <>
+struct ScalarOp<ONNXGeluOp> {
+  using FOp = CustomScalarOp;
+  using IOp = CustomScalarOp;
+};
+
+template <>
+double analyzeSimdFor<ONNXGeluOp>(
+    Type t, Operation *op, int64_t &von, int64_t &son) {
+  double results;
+  StringRef approximate = dyn_cast<ONNXGeluOp>(op).getApproximate();
+  if (approximate == "none")
+    results = simdAnalysis({GenericOps::ArithmeticGop, GenericOps::DivGop,
+                               GenericOps::MulGop, GenericOps::SqrtGop},
+        {1, 1, 2, 1}, t, von, son);
+  if (approximate == "tanh")
+    results = simdAnalysis(
+        {GenericOps::ArithmeticGop, GenericOps::DivGop, GenericOps::MulGop,
+            GenericOps::PowGop, GenericOps::SqrtGop},
+        {2, 1, 4, 1, 1}, t, von, son);
+  return results;
+}
+
+template <>
+Value emitScalarOpFor<ONNXGeluOp>(ConversionPatternRewriter &rewriter,
+    Location loc, Operation *op, Type elementType,
+    ArrayRef<Value> scalarOperands) {
+  Value operand = scalarOperands[0];
+
+  CheckIfCustomScalarOpIsSupported<ONNXGeluOp>(elementType);
+  MultiDialectBuilder<KrnlBuilder, MathBuilder> create(rewriter, loc);
+
+  // Approximate is a required attribute and should have a default value of
+  // "none". "approximate = none" simply implies no approximation will take
+  // place. However, "approximate" can also have a string value of "tanh" which
+  // indicates the use of tanh approximation.
+  StringRef approximate = dyn_cast<ONNXGeluOp>(op).getApproximate();
+
+  // Create constants
+  Value half = create.math.constant(elementType, 0.5);
+  Value one = create.math.constant(elementType, 1);
+  Value three = create.math.constant(elementType, 3);
+  Value decimal = create.math.constant(elementType, 0.044715);
+  Value sqrt_two_over_pi = create.math.constant(elementType, sqrt(2 / M_PI));
+  Value sqrt_two = create.math.constant(elementType, sqrt(2));
```

Comment:
Please use camel case for variable names, i.e. `sqrtTwoOverPi`, `sqrtTwo`, etc.

Reply:
Okay fixed!